pkg/api/types.go

```diff
@@ -123,11 +123,32 @@ type InputConfiguration struct {
 	// build-time dependencies for the project.
 	TestBaseImage *ImageStreamTagReference `json:"test_base_image,omitempty"`
 
+	// BuildRootImage supports two ways to get the image that
+	// the pipeline will caches on. The one way is to take the reference
+	// from an image stream, and the other from a dockerfile.
+	BuildRootImage *BuildRootImageConfiguration `json:"build_root,omitempty"`
+
 	// ReleaseTagConfiguration determines how the
 	// full release is assembled.
 	ReleaseTagConfiguration *ReleaseTagConfiguration `json:"tag_specification,omitempty"`
 }
 
+// BuildRootImageConfiguration holds the two ways of using a base image
+// that the pipeline will caches on.
+type BuildRootImageConfiguration struct {
```

Comment:
What is the exact use case this is solving?  Why is this is any different than any other "image"?

Reply:
Trying to give to user two options in how he will pass the base image. And the config can be like this:
```
{
  "build_root": {
    "git_source_image":{
      "dockerfile": "Dockerfile"
    }, 
or 
    "image_stream_tag":{
      "cluster": "https://api.ci.openshift.org",
      "namespace": "openshift",
      "name": "release",
      "tag": "golang-1.10"
    }
  }
}
```

If the Uri is not specified as the example above, it will get it from $JOB_SPEC